include/deal.II/matrix_free/shape_info.templates.h

```diff
@@ -238,6 +297,169 @@ namespace internal
     {
       static_assert(dim == spacedim,
                     "Currently, only the case dim=spacedim is implemented");
+
+      // ShapeInfo for RT elements. Here, data is of size 2 instead of 1.
+      // data[0] is univariate_shape_data in normal direction and
+      // data[1] is univariate_shape_data in tangential direction
+      //
+      if (dynamic_cast<const FE_RaviartThomasNodal<dim> *>(
+            &fe_in.base_element(base_element_number)))
+        {
+          element_type = tensor_raviart_thomas;
+
+          const auto quad = quad_in.get_tensor_basis()[0];
+
+          const FiniteElement<dim> &fe =
+            fe_in.base_element(base_element_number);
+          n_dimensions = dim;
+          n_components = fe_in.n_components();
+
+          data.resize(2);
+
+          // data_access.reinit(n_dimensions, n_components); // what are these?
+          // data_access.fill(&univariate_shape_data);
+
+          if ((fe.n_dofs_per_cell() == 0) || (quad.size() == 0))
+            return; // When do we want to check this?
+
+          const unsigned int n_q_points_1d = quad.size();
+
+          n_q_points      = Utilities::fixed_power<dim>(n_q_points_1d);
+          n_q_points_face = Utilities::fixed_power<dim - 1>(n_q_points_1d);
+
+          dofs_per_component_on_cell = fe_in.n_dofs_per_cell() / n_components;
+
+          // NOTE dofs_per_component_on_face is in tangential direction!
+          dofs_per_component_on_face =
+            fe_in.n_dofs_per_face() + Utilities::pow(fe_in.degree, dim - 2);
+          const unsigned int dofs_per_face_normal = fe_in.n_dofs_per_face();
+
+          lexicographic_numbering =
+            get_lexicographic_numbering_raviart_thomas<dim>(
+              fe_in.degree, fe_in.n_dofs_per_face());
+
+          // To get the right shape_values of the RT element
+          std::vector<unsigned int> lex_normal, lex_tangent;
+          for (unsigned int i = 0; i < fe.degree; i++)
+            lex_tangent.push_back(i);
+
+          lex_normal.push_back(0);
+          for (unsigned int i = dofs_per_face_normal * 2 * dim;
+               i < dofs_per_face_normal * 2 * dim + fe.degree - 1;
+               i++)
+            lex_normal.push_back(i);
+          lex_normal.push_back(dofs_per_face_normal);
```

Comment:
I am wondering if wouldn't be easier to setup a temporal 2D RT element? `dim` looks like an additional complexity here.